app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -479,6 +483,17 @@ def remove_lifecycle_environment(environment)
         raise _("Lifecycle environment was not attached to the smart proxy; therefore, no changes were made.")
       end
 
+      def refresh_smart_proxy_sync_histories
+        smart_proxy_helper = ::Katello::SmartProxyHelper.new(self)
+        repos = smart_proxy_helper.repositories_available_to_capsule
+        repo_ids = repos.map(&:id)
+        if repo_ids.empty?
+          self.smart_proxy_sync_histories.delete_all
+        else
+          self.smart_proxy_sync_histories.where("repository_id NOT IN (?)", repo_ids).delete_all
+        end
```

Comment:
```suggestion
        repos = smart_proxy_helper.repositories_available_to_capsule.select(:id)
        # Use size to ensure repos don't get loaded unnecessarily
        if repos.size == 0
          self.smart_proxy_sync_histories.delete_all
        else
          # Uses a single SQL query without needing to load IDs
          self.smart_proxy_sync_histories.where(id: repos).delete_all
        end
```

Reply:
Ack..Updated to this with slight change to maintain previous logic: 
`self.smart_proxy_sync_histories.where.not(repository_id: repos).delete_all`